### PR TITLE
Add a new word-based clock app

### LINKED ISSUE #179: App Idea: Word Clock
PR to follow

### PR DESCRIPTION
**Pull request by bmsleight**
Signed-off-by: Brendan M. Sleight <bms.git@barwap.com>

### CODE REVIEW

**Comment by bmsleight** [comment]
Any more comments ?
Merge or not merge ?

File: wasp/apps/word_clock.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+# Copyright (C) 2020 Daniel Thompson
+# Copyright (C) 2021 Brendan Sleight
+
+
+"""Digital clock
```
Comment (daniel-thompson) [inline wasp/apps/word_clock.py:6]: This looks out of date. It needs to be different from the other existing clocks.

File: wasp/apps/word_clock.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        """Draw or lazily update the display.
+
+        The updates are as lazy by default and avoid spending time redrawing
+        if the time on display has not changed. However if redraw is set to
+        True then a full redraw is be performed.
+        """
+        draw = wasp.watch.drawable
+        hi =  wasp.system.theme('bright')
+        lo =  wasp.system.theme('mid')
+        mid = draw.lighten(lo, 1)
```
Comment (daniel-thompson) [inline wasp/apps/word_clock.py:68]: Are lo and mid used anywhere?

File: wasp/apps/word_clock.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        self._words = self._words + hour + "\n" 
+        if (part_day !=""):
+            self._words = self._words + part_day 
+        
+        # Some phases may be 5 lines long, some may be 1 
+        draw.fill(0, 0, 48)
+
+        chunks = draw.wrap(self._words, 240)
+        lines_of_text = len(chunks)-1
+        offset_y=int(((5-lines_of_text/2)*24))
```
Comment (daniel-thompson) [inline wasp/apps/word_clock.py:186]: Perhaps change the * 24 to something slightly more spaced (26? 27?) to give a better appearance.

File: wasp/apps/word_clock.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+        hours_a = ["Midnight", "One", "Two", 
+                "Three", "Four", "Five", 
+                "Six", "Seven", "Eight", 
+                "Nine", "Ten", "Eleven", 
+                "Twelve",
+                "One", "Two", 
+                "Three", "Four", "Five", 
+                "Six", "Seven", "Eight", 
+                "Nine", "Ten", "Eleven"]
```
Comment (daniel-thompson) [inline wasp/apps/word_clock.py:116]: Why it this capitalized. Is it possible to capitalize the first word of the phrase instead?

**Review by daniel-thompson — CHANGES_REQUESTED** [review]
Generally this looks ok, just a few niggles really.

**Comment by daniel-thompson** [comment]
Applied. Thanks!

**Pull request closed without merging**